File: apps/pos/native-android/app/src/main/java/com/desktopkitchen/pos/viewmodels/ReportsViewModel.kt
package com.desktopkitchen.pos.viewmodels

import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.setValue
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.desktopkitchen.pos.models.HourlyReport
import com.desktopkitchen.pos.models.SalesReport
import com.desktopkitchen.pos.models.TopItemsReport
import com.desktopkitchen.pos.services.ReportService
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.launch
import javax.inject.Inject

enum class ReportPeriod(val value: String, val label: String) {
    TODAY("today", "Today"),
    WEEK("week", "This Week"),
    MONTH("month", "This Month")
}

@HiltViewModel
class ReportsViewModel @Inject constructor(
    private val reportService: ReportService
) : ViewModel() {

    var period by mutableStateOf(ReportPeriod.TODAY)
    var salesData by mutableStateOf<SalesReport?>(null)
        private set
    var topItems by mutableStateOf<List<TopItemsReport>>(emptyList())
        private set
    var hourlyData by mutableStateOf<List<HourlyReport>>(emptyList())
        private set
    var isLoading by mutableStateOf(true)
        private set
    var error by mutableStateOf<String?>(null)
        private set

    fun loadData() {
        viewModelScope.launch {
            isLoading = true
            error = null
            try {
                salesData = reportService.getSales(period.value)
                topItems = reportService.getTopItems(period.value, 10)
                hourlyData = reportService.getHourly()
            } catch (e: Exception) {
                error = e.message
            }
            isLoading = false
        }
    }

    fun changePeriod(newPeriod: ReportPeriod) {
        period = newPeriod
        loadData()
    }
}

File: apps/pos/native-windows/DesktopKitchenPOS/ViewModels/ReportsViewModel.cs
using CommunityToolkit.Mvvm.ComponentModel;
using CommunityToolkit.Mvvm.Input;
using DesktopKitchenPOS.Models;

namespace DesktopKitchenPOS.ViewModels;

public enum ReportPeriod
{
    Today,
    Week,
    Month
}

public partial class ReportsViewModel : ObservableObject
{
    [ObservableProperty]
    private ReportPeriod _period = ReportPeriod.Today;

    [ObservableProperty]
    private SalesReport? _salesData;

    [ObservableProperty]
    private List<TopItemsReport> _topItems = new();

    [ObservableProperty]
    private List<HourlyReport> _hourlyData = new();

    [ObservableProperty]
    private bool _isLoading = true;

    [ObservableProperty]
    private string? _error;

    public string PeriodValue => Period switch
    {
        ReportPeriod.Today => "today",
        ReportPeriod.Week => "week",
        ReportPeriod.Month => "month",
        _ => "today"
    };

    public string PeriodLabel => Period switch
    {
        ReportPeriod.Today => "Today",
        ReportPeriod.Week => "This Week",
        ReportPeriod.Month => "This Month",
        _ => "Today"
    };

    [RelayCommand]
    public async Task LoadDataAsync()
    {
        IsLoading = true;
        Error = null;
        try
        {
            var salesTask = DesktopKitchenPOS.App.ReportService.GetSalesAsync(PeriodValue);
            var itemsTask = DesktopKitchenPOS.App.ReportService.GetTopItemsAsync(PeriodValue, 10);
            var hourlyTask = DesktopKitchenPOS.App.ReportService.GetHourlyAsync();

            SalesData = await salesTask;
            TopItems = await itemsTask;
            HourlyData = await hourlyTask;
        }
        catch (Exception ex)
        {
            Error = ex.Message;
        }
        IsLoading = false;
    }

    [RelayCommand]
    public async Task ChangePeriodAsync(ReportPeriod newPeriod)
    {
        Period = newPeriod;
        await LoadDataAsync();
    }
}
